diff --git a/Source/NSMenuView.m b/Source/NSMenuView.m
--- a/Source/NSMenuView.m
+++ b/Source/NSMenuView.m
@@ -1832,11 +1832,10 @@
 					       dequeue: YES];
 	  type = [event type];
     if (type == NSLeftMouseUp || type == NSRightMouseUp || type == NSOtherMouseUp)
       {
-          shouldFinish = YES;
           break;  // Exit the loop to proceed to StopPeriodicEvents
       }
 	  if (type == NSAppKitDefined)
 	    {
 	      [[event window] sendEvent: event];
 	    }
